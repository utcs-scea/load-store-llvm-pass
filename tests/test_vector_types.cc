#include <stdio.h>

int global_array[5] = {1, 2, 3, 4, 5};
int *global_ptr = global_array;

void modify_array() {
  for (int i = 0; i < 5; i++) {
    global_array[i] *= 2;
  }
}

int main() {
  printf("Initial array: ");
  for (int i = 0; i < 5; i++) {
    printf("%d ", global_ptr[i]);
  }
  printf("\n");

  modify_array();

  printf("Modified array: ");
  for (int i = 0; i < 5; i++) {
    printf("%d ", global_array[i]);
  }
  printf("\n");

  return 0;
}
