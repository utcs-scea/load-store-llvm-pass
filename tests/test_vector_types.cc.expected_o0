   >> globalify() invoked
   >> globalify() invoked
   >> globalify() invoked
   >> __pando__replace_store_int32() invoked
   >> check_if_global() invoked
   >> deglobalify() invoked
   >> globalify() invoked
   >> deglobalify() invoked
Initial array: 
   >> __pando__replace_store_int32() invoked
   >> check_if_global() invoked
   >> deglobalify() invoked
   >> __pando__replace_load_int32() invoked
   >> check_if_global() invoked
   >> deglobalify() invoked
   >> globalify() invoked
   >> __pando__replace_load_ptr() invoked
   >> check_if_global() invoked
   >> deglobalify() invoked
   >> globalify() invoked
   >> __pando__replace_load_int32() invoked
   >> check_if_global() invoked
   >> deglobalify() invoked
   >> __pando__replace_load_int32() invoked
   >> check_if_global() invoked
   >> deglobalify() invoked
   >> globalify() invoked
   >> deglobalify() invoked
1 
   >> __pando__replace_load_int32() invoked
   >> check_if_global() invoked
   >> deglobalify() invoked
   >> __pando__replace_store_int32() invoked
   >> check_if_global() invoked
   >> deglobalify() invoked
   >> __pando__replace_load_int32() invoked
   >> check_if_global() invoked
   >> deglobalify() invoked
   >> globalify() invoked
   >> __pando__replace_load_ptr() invoked
   >> check_if_global() invoked
   >> deglobalify() invoked
   >> globalify() invoked
   >> __pando__replace_load_int32() invoked
   >> check_if_global() invoked
   >> deglobalify() invoked
   >> __pando__replace_load_int32() invoked
   >> check_if_global() invoked
   >> deglobalify() invoked
   >> globalify() invoked
   >> deglobalify() invoked
2 
   >> __pando__replace_load_int32() invoked
   >> check_if_global() invoked
   >> deglobalify() invoked
   >> __pando__replace_store_int32() invoked
   >> check_if_global() invoked
   >> deglobalify() invoked
   >> __pando__replace_load_int32() invoked
   >> check_if_global() invoked
   >> deglobalify() invoked
   >> globalify() invoked
   >> __pando__replace_load_ptr() invoked
   >> check_if_global() invoked
   >> deglobalify() invoked
   >> globalify() invoked
   >> __pando__replace_load_int32() invoked
   >> check_if_global() invoked
   >> deglobalify() invoked
   >> __pando__replace_load_int32() invoked
   >> check_if_global() invoked
   >> deglobalify() invoked
   >> globalify() invoked
   >> deglobalify() invoked
3 
   >> __pando__replace_load_int32() invoked
   >> check_if_global() invoked
   >> deglobalify() invoked
   >> __pando__replace_store_int32() invoked
   >> check_if_global() invoked
   >> deglobalify() invoked
   >> __pando__replace_load_int32() invoked
   >> check_if_global() invoked
   >> deglobalify() invoked
   >> globalify() invoked
   >> __pando__replace_load_ptr() invoked
   >> check_if_global() invoked
   >> deglobalify() invoked
   >> globalify() invoked
   >> __pando__replace_load_int32() invoked
   >> check_if_global() invoked
   >> deglobalify() invoked
   >> __pando__replace_load_int32() invoked
   >> check_if_global() invoked
   >> deglobalify() invoked
   >> globalify() invoked
   >> deglobalify() invoked
4 
   >> __pando__replace_load_int32() invoked
   >> check_if_global() invoked
   >> deglobalify() invoked
   >> __pando__replace_store_int32() invoked
   >> check_if_global() invoked
   >> deglobalify() invoked
   >> __pando__replace_load_int32() invoked
   >> check_if_global() invoked
   >> deglobalify() invoked
   >> globalify() invoked
   >> __pando__replace_load_ptr() invoked
   >> check_if_global() invoked
   >> deglobalify() invoked
   >> globalify() invoked
   >> __pando__replace_load_int32() invoked
   >> check_if_global() invoked
   >> deglobalify() invoked
   >> __pando__replace_load_int32() invoked
   >> check_if_global() invoked
   >> deglobalify() invoked
   >> globalify() invoked
   >> deglobalify() invoked
5 
   >> __pando__replace_load_int32() invoked
   >> check_if_global() invoked
   >> deglobalify() invoked
   >> __pando__replace_store_int32() invoked
   >> check_if_global() invoked
   >> deglobalify() invoked
   >> __pando__replace_load_int32() invoked
   >> check_if_global() invoked
   >> deglobalify() invoked
   >> globalify() invoked
   >> __pando__replace_store_int32() invoked
   >> check_if_global() invoked
   >> deglobalify() invoked
   >> __pando__replace_load_int32() invoked
   >> check_if_global() invoked
   >> deglobalify() invoked
   >> __pando__replace_load_int32() invoked
   >> check_if_global() invoked
   >> deglobalify() invoked
   >> globalify() invoked
   >> __pando__replace_load_int32() invoked
   >> check_if_global() invoked
   >> deglobalify() invoked
   >> __pando__replace_store_int32() invoked
   >> check_if_global() invoked
   >> deglobalify() invoked
   >> __pando__replace_load_int32() invoked
   >> check_if_global() invoked
   >> deglobalify() invoked
   >> __pando__replace_store_int32() invoked
   >> check_if_global() invoked
   >> deglobalify() invoked
   >> __pando__replace_load_int32() invoked
   >> check_if_global() invoked
   >> deglobalify() invoked
   >> __pando__replace_load_int32() invoked
   >> check_if_global() invoked
   >> deglobalify() invoked
   >> globalify() invoked
   >> __pando__replace_load_int32() invoked
   >> check_if_global() invoked
   >> deglobalify() invoked
   >> __pando__replace_store_int32() invoked
   >> check_if_global() invoked
   >> deglobalify() invoked
   >> __pando__replace_load_int32() invoked
   >> check_if_global() invoked
   >> deglobalify() invoked
   >> __pando__replace_store_int32() invoked
   >> check_if_global() invoked
   >> deglobalify() invoked
   >> __pando__replace_load_int32() invoked
   >> check_if_global() invoked
   >> deglobalify() invoked
   >> __pando__replace_load_int32() invoked
   >> check_if_global() invoked
   >> deglobalify() invoked
   >> globalify() invoked
   >> __pando__replace_load_int32() invoked
   >> check_if_global() invoked
   >> deglobalify() invoked
   >> __pando__replace_store_int32() invoked
   >> check_if_global() invoked
   >> deglobalify() invoked
   >> __pando__replace_load_int32() invoked
   >> check_if_global() invoked
   >> deglobalify() invoked
   >> __pando__replace_store_int32() invoked
   >> check_if_global() invoked
   >> deglobalify() invoked
   >> __pando__replace_load_int32() invoked
   >> check_if_global() invoked
   >> deglobalify() invoked
   >> __pando__replace_load_int32() invoked
   >> check_if_global() invoked
   >> deglobalify() invoked
   >> globalify() invoked
   >> __pando__replace_load_int32() invoked
   >> check_if_global() invoked
   >> deglobalify() invoked
   >> __pando__replace_store_int32() invoked
   >> check_if_global() invoked
   >> deglobalify() invoked
   >> __pando__replace_load_int32() invoked
   >> check_if_global() invoked
   >> deglobalify() invoked
   >> __pando__replace_store_int32() invoked
   >> check_if_global() invoked
   >> deglobalify() invoked
   >> __pando__replace_load_int32() invoked
   >> check_if_global() invoked
   >> deglobalify() invoked
   >> __pando__replace_load_int32() invoked
   >> check_if_global() invoked
   >> deglobalify() invoked
   >> globalify() invoked
   >> __pando__replace_load_int32() invoked
   >> check_if_global() invoked
   >> deglobalify() invoked
   >> __pando__replace_store_int32() invoked
   >> check_if_global() invoked
   >> deglobalify() invoked
   >> __pando__replace_load_int32() invoked
   >> check_if_global() invoked
   >> deglobalify() invoked
   >> __pando__replace_store_int32() invoked
   >> check_if_global() invoked
   >> deglobalify() invoked
   >> __pando__replace_load_int32() invoked
   >> check_if_global() invoked
   >> deglobalify() invoked
   >> globalify() invoked
   >> deglobalify() invoked
Modified array: 
   >> __pando__replace_store_int32() invoked
   >> check_if_global() invoked
   >> deglobalify() invoked
   >> __pando__replace_load_int32() invoked
   >> check_if_global() invoked
   >> deglobalify() invoked
   >> __pando__replace_load_int32() invoked
   >> check_if_global() invoked
   >> deglobalify() invoked
   >> globalify() invoked
   >> __pando__replace_load_int32() invoked
   >> check_if_global() invoked
   >> deglobalify() invoked
   >> globalify() invoked
   >> deglobalify() invoked
2 
   >> __pando__replace_load_int32() invoked
   >> check_if_global() invoked
   >> deglobalify() invoked
   >> __pando__replace_store_int32() invoked
   >> check_if_global() invoked
   >> deglobalify() invoked
   >> __pando__replace_load_int32() invoked
   >> check_if_global() invoked
   >> deglobalify() invoked
   >> __pando__replace_load_int32() invoked
   >> check_if_global() invoked
   >> deglobalify() invoked
   >> globalify() invoked
   >> __pando__replace_load_int32() invoked
   >> check_if_global() invoked
   >> deglobalify() invoked
   >> globalify() invoked
   >> deglobalify() invoked
4 
   >> __pando__replace_load_int32() invoked
   >> check_if_global() invoked
   >> deglobalify() invoked
   >> __pando__replace_store_int32() invoked
   >> check_if_global() invoked
   >> deglobalify() invoked
   >> __pando__replace_load_int32() invoked
   >> check_if_global() invoked
   >> deglobalify() invoked
   >> __pando__replace_load_int32() invoked
   >> check_if_global() invoked
   >> deglobalify() invoked
   >> globalify() invoked
   >> __pando__replace_load_int32() invoked
   >> check_if_global() invoked
   >> deglobalify() invoked
   >> globalify() invoked
   >> deglobalify() invoked
6 
   >> __pando__replace_load_int32() invoked
   >> check_if_global() invoked
   >> deglobalify() invoked
   >> __pando__replace_store_int32() invoked
   >> check_if_global() invoked
   >> deglobalify() invoked
   >> __pando__replace_load_int32() invoked
   >> check_if_global() invoked
   >> deglobalify() invoked
   >> __pando__replace_load_int32() invoked
   >> check_if_global() invoked
   >> deglobalify() invoked
   >> globalify() invoked
   >> __pando__replace_load_int32() invoked
   >> check_if_global() invoked
   >> deglobalify() invoked
   >> globalify() invoked
   >> deglobalify() invoked
8 
   >> __pando__replace_load_int32() invoked
   >> check_if_global() invoked
   >> deglobalify() invoked
   >> __pando__replace_store_int32() invoked
   >> check_if_global() invoked
   >> deglobalify() invoked
   >> __pando__replace_load_int32() invoked
   >> check_if_global() invoked
   >> deglobalify() invoked
   >> __pando__replace_load_int32() invoked
   >> check_if_global() invoked
   >> deglobalify() invoked
   >> globalify() invoked
   >> __pando__replace_load_int32() invoked
   >> check_if_global() invoked
   >> deglobalify() invoked
   >> globalify() invoked
   >> deglobalify() invoked
10 
   >> __pando__replace_load_int32() invoked
   >> check_if_global() invoked
   >> deglobalify() invoked
   >> __pando__replace_store_int32() invoked
   >> check_if_global() invoked
   >> deglobalify() invoked
   >> __pando__replace_load_int32() invoked
   >> check_if_global() invoked
   >> deglobalify() invoked
