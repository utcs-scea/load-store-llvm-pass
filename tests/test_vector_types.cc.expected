   >> globalify() invoked
   >> globalify() invoked
   >> deglobalify() invoked
Initial array: 
   >> globalify() invoked
   >> __pando__replace_load_ptr() invoked
   >> check_if_global() invoked
   >> deglobalify() invoked
   >> __pando__replace_load_int32() invoked
   >> check_if_global() invoked
   >> deglobalify() invoked
   >> globalify() invoked
   >> deglobalify() invoked
1 
   >> globalify() invoked
   >> __pando__replace_load_ptr() invoked
   >> check_if_global() invoked
   >> deglobalify() invoked
   >> __pando__replace_load_int32() invoked
   >> check_if_global() invoked
   >> deglobalify() invoked
   >> globalify() invoked
   >> deglobalify() invoked
2 
   >> globalify() invoked
   >> __pando__replace_load_ptr() invoked
   >> check_if_global() invoked
   >> deglobalify() invoked
   >> __pando__replace_load_int32() invoked
   >> check_if_global() invoked
   >> deglobalify() invoked
   >> globalify() invoked
   >> deglobalify() invoked
3 
   >> globalify() invoked
   >> __pando__replace_load_ptr() invoked
   >> check_if_global() invoked
   >> deglobalify() invoked
   >> __pando__replace_load_int32() invoked
   >> check_if_global() invoked
   >> deglobalify() invoked
   >> globalify() invoked
   >> deglobalify() invoked
4 
   >> globalify() invoked
   >> __pando__replace_load_ptr() invoked
   >> check_if_global() invoked
   >> deglobalify() invoked
   >> __pando__replace_load_int32() invoked
   >> check_if_global() invoked
   >> deglobalify() invoked
   >> globalify() invoked
   >> deglobalify() invoked
5 
   >> globalify() invoked
  >> __pando__replace_load_vector invoked
   >> check_if_global() invoked
   >> deglobalify() invoked
   >> globalify() invoked
  >> __pando__replace_store_vector invoked
   >> check_if_global() invoked
   >> deglobalify() invoked
   >> globalify() invoked
   >> __pando__replace_load_int32() invoked
   >> check_if_global() invoked
   >> deglobalify() invoked
   >> globalify() invoked
   >> __pando__replace_store_int32() invoked
   >> check_if_global() invoked
   >> deglobalify() invoked
   >> globalify() invoked
   >> deglobalify() invoked
Modified array: 
   >> globalify() invoked
   >> __pando__replace_load_int32() invoked
   >> check_if_global() invoked
   >> deglobalify() invoked
   >> globalify() invoked
   >> deglobalify() invoked
2 
   >> globalify() invoked
   >> __pando__replace_load_int32() invoked
   >> check_if_global() invoked
   >> deglobalify() invoked
   >> globalify() invoked
   >> deglobalify() invoked
4 
   >> globalify() invoked
   >> __pando__replace_load_int32() invoked
   >> check_if_global() invoked
   >> deglobalify() invoked
   >> globalify() invoked
   >> deglobalify() invoked
6 
   >> globalify() invoked
   >> __pando__replace_load_int32() invoked
   >> check_if_global() invoked
   >> deglobalify() invoked
   >> globalify() invoked
   >> deglobalify() invoked
8 
   >> globalify() invoked
   >> __pando__replace_load_int32() invoked
   >> check_if_global() invoked
   >> deglobalify() invoked
   >> globalify() invoked
   >> deglobalify() invoked
10 
